elasticsearch/serializer.py

```diff
@@ -27,12 +63,29 @@ class JSONSerializer(object):
     mimetype = "application/json"
 
     def default(self, data):
-        if isinstance(data, (date, datetime)):
+        if isinstance(data, TIME_TYPES):
             return data.isoformat()
-        elif isinstance(data, Decimal):
-            return float(data)
         elif isinstance(data, uuid.UUID):
             return str(data)
+        elif isinstance(data, FLOAT_TYPES):
+            return float(data)
+        elif INTEGER_TYPES and isinstance(data, INTEGER_TYPES):
+            return int(data)
+
+        # Special cases for numpy and pandas types
+        elif np:
+            if isinstance(data, np.bool_):
+                return bool(data)
+            elif isinstance(data, np.datetime64):
+                return data.item().isoformat()
+            elif isinstance(data, np.ndarray):
+                return data.tolist()
+        if pd:
+            if isinstance(data, pd.Series):
```

Comment:
I'm definitely not the expert here (so I might asking the wrong question here), but I've been recently working a bit with `pandas.dtypes` and I'm wondering, how will this serialiser handle the types of the elements inside a `pd.Series` or a `pd.DataFrame`. Usually these elements are various numerical types, so probably handled by the Numpy converters, but sometimes a user can set these dtypes to pandas specific stuff like `category`. For ex. I've recently done something like this in my Jupyter notebook

```
# finally we will correct the mappings on the remaining columns 

mappings = {'carat': 'float64',
           'cut': 'category',
           'color': 'category',
           'depth':'float64',
           'table':'float64',
           'price': 'float64',
           'x':'float64',
           'y':'float64',
           'z':'float64'}

df_cleaned = df_cleaned.astype(mappings)
```

How would the serialisation handle `category` or some of the other pandas specific dtypes in a Series (here is a list of some more exotic dtypes https://pandas.pydata.org/pandas-docs/stable/getting_started/basics.html#dtypes)?

Reply:
Category is an interesting one, we'd lose the "categorical" aspect of the value if we serialize to a string but maybe that's fine? Maintaining the categorical aspect would require config on the mapping but as long as that's done it'd be a solution, so maybe we do `category` -> `str`?